packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,287 @@
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      resource.iamServiceRole = role.roleArn;
+    }
+
+    this.appArn = resource.attrArn,
+    this.appDefaultDomain = resource.attrDefaultDomain;
+    this.appId = resource.attrAppId;
+    this.appName = resource.attrAppName;
+
+    this.resource = resource;
+  }
+
+  public validate(): string[] {
+    const errors: string[] = [];
+
+    if (this.resource.buildSpec && !this.resource.iamServiceRole) {
+      errors.push(`You need to specify a service role when using buildspecs`);
+    }
+
+    return errors;
+  }
+
+  /**
+   * Add Service Role
+   *
+   * @param role
+   */
+  public addServiceRole(role: IRole) {
+    this.resource.iamServiceRole = role.roleArn;
+  }
+
+  /**
+   * Set Basic Auth on App
+   *
+   * @param username
+   * @param password
+   */
+  public setBasicAuth(username: string, password: string) {
+    this.basicAuthResolver.basicAuthConfig({
+      enableBasicAuth: true,
+      password,
+      username
+    });
+  }
+
+  /**
+   * Add Environment Variable to App
+   *
+   * @param name
+   * @param value
+   */
+  public addEnvironmentVariable(name: string, value: string) {
+    this.environmentVariablesResolver.addEnvironmentVariable(name, value);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Options
+ */
+export interface AppOptions extends ResourceProps {
+  /**
+   * App Name
+   *
+   * @todo This might cause an error, it conflicts with the construct
+   *
+   * @default
+   */
+  readonly appName?: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps extends AppOptions {
+  /**
+   * Access Token
+   *
+   * @default
+   */
+  readonly accessToken?: string;
+
+  /**
+   * Basic Auth Config
+   *
+   * @default
+   */
+  readonly basicAuthConfig?: BasicAuthConfig;
+
+  /**
+   * BuildSpec
+   *
+   * @default
+   */
+  readonly buildSpec?: string;
+
+  /**
+   * Custom Rules
+   *
+   * @default
+   */
+  readonly customRules?: CustomRule[];
+
+  /**
+   * Description
+   *
+   * @default
+   */
+  readonly description?: string;
+
+  /**
+   * Environment Variables
+   *
+   * @default
+   */
+  readonly environmentVariables?: EnvironmentVariable[];
+
+  /**
+   * IAM Service Role
+   *
+   * @default
+   */
+  readonly iamServiceRole?: IRole;
+
+  /**
+   * Name
+   */
+  readonly name: string;
+
+  /**
+   * Name
+   *
+   * @default
+   */
+  readonly oauthToken?: string;
+
+  /**
+   * Repository
+   *
+   * Required to be in HTTP URL format: https://github.com/awslabs/aws-cdk
+   * If you use a Git URL, it will error.
+   */
+  readonly repository: string;
+
+  /**
+   * Tags
+   *
+   * @default
+   */
+  readonly tags?: Tag[];
+}
+
+/**
+ * Custom Rule Interface
+ */
+export interface CustomRule {
+  /**
+   * Condition
+   *
+   * @default Empty
```

Comment:
Dis it the string “Empty” or is it “- all rules match”. Always add a description to the default value the explains the behavior and use “-“ to not specify a value

Reply:
Actually empty/undefined